echoregions/lines/lines.py

```diff
@@ -212,42 +250,119 @@ def filter_bottom(bottom, start_date, end_date):
         # filter bottom within start and end time
         start_time = da_Sv.ping_time.data.min()
         end_time = da_Sv.ping_time.data.max()
-        filtered_bottom = filter_bottom(lines_df, start_time, end_time)
+        filtered_bottom = self._filter_bottom(lines_df, start_time, end_time, operation)
         filtered_bottom = filtered_bottom[~filtered_bottom.index.duplicated()]
 
         if len(filtered_bottom) > 0:
-            # create joint index
-            joint_index = list(
-                set(list(pd.DataFrame(echogram_ping_time)[0]) + list(filtered_bottom.index))
-            )
-
-            # Interpolate on the echogram coordinates. Note that some interpolation kwaargs
-            # will result in some interpolation NaN values. The ffill and bfill lines
-            # are there to fill in these NaN values.
-            # TODO There exists a problem where when we use .loc prior to reindexing
-            # we are hit with a key not found error.
-            bottom_points = (
-                filtered_bottom[["depth"]]
-                .reindex(joint_index)
-                .interpolate(**kwargs)
-                .loc[echogram_ping_time]
-                .ffill()
-                .bfill()
-            )
-
-            # convert to data array for bottom
-            bottom_da = bottom_points["depth"].to_xarray()
-            bottom_da = bottom_da.rename({"time": "ping_time"})
-
-            # create a data array of depths
-            depth_da = da_Sv["depth"] + xr.zeros_like(da_Sv)
-
-            # create a mask for the bottom:
-            # bottom: True, otherwise: False
-            bottom_mask = depth_da >= bottom_da
-
-            # Reset bottom_points index so that time index becomes time column
-            bottom_points = bottom_points.reset_index()
+            if operation == "regionmask":
+                # Filter columns and sort rows
+                bottom_points = filtered_bottom.copy()[["time", "depth"]].sort_values(by="time")
+
+                # Calculate left and right most bottom point depth values
+                bottom_points_min_time_depth = bottom_points.loc[
+                    bottom_points["time"].idxmin(), "depth"
+                ]
+                bottom_points_max_time_depth = bottom_points.loc[
+                    bottom_points["time"].idxmax(), "depth"
+                ]
+
+                # Calculate maximum depth between bottom points and Sv and add additional offset
+                maximum_depth_plus_offset = (
+                    max([da_Sv["depth"].max().data, bottom_points["depth"].max()]) + 50.0
```

Comment:
Why do you have an offset here? Can you add a little comment in the code so that it is clear?

Reply:
So the offset is here to make sure that the line connecting the bottom-most points are clear of any other points. This would be a problem in the case where there is a point in the middle that matches the maximum of the Sv and EVL point depth. This would lead to regionmask creating possibly 2+ regions, which is behavior that could lead to different outputs. The offset ensures that regionmask always creates just 1 region. 50 isn't really needed here, I could just use 1.  I'll add a comment to make this more clear.